src/test/java/io/cdap/plugin/aws/s3/source/S3BatchSourceTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.plugin.aws.s3.source;
+
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.aws.s3.common.S3ConnectorConfig;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+/**
+ * Test class for S3 Batch Source
+ */
+public class S3BatchSourceTest {
+
+  @Test
+  public void testValidateWithoutCredentials() throws IllegalAccessException, NoSuchFieldException {
+    MockFailureCollector mockFailureCollector = new MockFailureCollector();
+    S3BatchSource.S3BatchConfig s3BatchConfig = new S3BatchSource.S3BatchConfig();
+    Field referenceNameField = s3BatchConfig.getClass().getSuperclass().getDeclaredField("referenceName");
```

Comment:
Don't set fields like this. It is brittle (renaming a field will break this test).

Create a Builder for the config and create it normally.

our normal pattern is to put the Builder within the class that it builds. see `GCSBatchSinkConfig.Builder`: https://github.com/data-integrations/google-cloud/blob/f75f09d80c77bfae444687c9b9749779315e7779/src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java#L646

Reply:
Added a configuration builder class for setting fields. 641656002bc887ea412645315d08b7b3f50ad8eb